### PR TITLE
Document the eventing setup in dogfooding

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Knative eventing + Kafka broken is now available in dogfooding, and it's
used to transport cloudevents generated by Tekton pipelines to
consumers.

Document the installation process, the knative triggers used to get
messages to consumers and alter the tekton pipeline configuration to
send events to the broker.

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

/kind misc

### CODE REVIEW

**Comment by afrittoli** [comment]
@dibyom DYI I documented the Kafka / Eventing setup :)

File: docs/dogfooding.md [hunk 1]
```diff
@@ -144,4 +169,159 @@ spec:
...
+data:
+  default-cloud-events-sink: http://kafka-broker-ingress.knative-eventing.svc.cluster.local/default/default
+```
+
+### CloudEvents Consumers
+
+`CloudEvents` are consumed from the broker via a Knative Eventing CRD called `Trigger`.
+The `dogfooding` cluster is setup so that all `TaskRun` start, running and finish events are forwarded from the
+broker to the `tekton-events` event listener, in the `default` namespace.
+This initial filtering of events allows to reduce the load on the event listener.
```
Comment (dibyom) [inline docs/dogfooding.md:271]: This is nice. Did we have performance issues due to heavy load on the EL?

**Review by dibyom — APPROVED** [review]
Thank you! This looks great

File: docs/dogfooding.md (hunk 1 above)
Comment (afrittoli) [reply docs/dogfooding.md:271]: I don't have hard numbers on this, but my feeling is that the el pod was coping OK with the load.
But I'm happy to reduce the noise in the logs and simplify the CEL filtering.

Comment (dibyom) [reply docs/dogfooding.md:271]: yeah there is definitely a lot of logs.

**Pull request merged**